sycl/source/detail/stream_impl.cpp

```diff
@@ -67,7 +67,7 @@ void stream_impl::flush() {
   // finishes execution.
   auto Q = detail::createSyclObjFromImpl<queue>(
       cl::sycl::detail::Scheduler::getInstance().getDefaultHostQueue());
-  Q.submit([&](handler &cgh) {
+  auto Event = Q.submit([&](handler &cgh) {
```

Comment:
> During this cleanup process stream buffer is deleted, as a result scheduler does a blocking enqueue of the empty task.

As far as I understand the cleanup may happen right after the kernel is executed while the printing host-task isn't executed yet. Is that correct?
Hence, I wonder if it is possible that `stream_impl::flush` is invoked after kernel's execution?